src/components/LocaleDateTime.tsx

```diff
@@ -0,0 +1,53 @@
+import { useRouter } from "next/router"
+
+type LocaleDateTimeProps = {
+  utcDateTime: string
+  hideDate?: boolean
+  hideTime?: boolean
+  options?: Intl.DateTimeFormatOptions
+}
+
+/**
+ * Renders a localized date and time component
+ * @param utcDateTime - The UTC date and time string, ie "2022-04-20T16:20:00Z"
+ * @param hideDate - Whether to hide the date portion
```

Comment:
Do we have cases where we just want to display date and not time or viceversa?

Reply:
Throwing an error will occur at build, not for the user. This prevents accidentally hiding the entire date and time.

Hiding one or the other can be desired, ie the Dencun date... we don't need to show the exact time, but depending which timezone you live in this will either occur on the 13th or 14th. That was why I placed "UTC" after the date to remove that ambiguity, but this component would just render it accordingly in the users local timezone (which doesn't just affect the time, but the date as well)